man/systemd.exec.xml

```diff
@@ -482,175 +482,181 @@
-        this parameter might result in additional dependencies to be
-        added to the unit (see above).</para></listitem>
+        <para>This setting defaults to the value set with <option>DefaultStandardOutput=</option> in
+        <citerefentry><refentrytitle>systemd-system.conf</refentrytitle><manvolnum>5</manvolnum></citerefentry>, which
+        defaults to <option>journal</option>. Note that setting this parameter might result in additional dependencies
+        to be added to the unit (see above).</para></listitem>
       </varlistentry>
 
       <varlistentry>
         <term><varname>StandardError=</varname></term>
-        <listitem><para>Controls where file descriptor 2 (STDERR) of
-        the executed processes is connected to. The available options
-        are identical to those of <varname>StandardOutput=</varname>,
-        with some exceptions: if set to <option>inherit</option> the
-        file descriptor used for standard output is duplicated for
-        standard error, while <option>fd</option> operates on the error
-        stream and will look by default for a descriptor named
+        <listitem><para>Controls where file descriptor 2 (STDERR) of the executed processes is connected to. The
+        available options are identical to those of <varname>StandardOutput=</varname>, with some exceptions: if set to
+        <option>inherit</option> the file descriptor used for standard output is duplicated for standard error, while
+        <option>fd:<replaceable>name</replaceable></option> will use a default file descriptor name of
         <literal>stderr</literal>.</para>
 
-        <para>This setting defaults to the value set with
-        <option>DefaultStandardError=</option> in
-        <citerefentry><refentrytitle>systemd-system.conf</refentrytitle><manvolnum>5</manvolnum></citerefentry>,
-        which defaults to <option>inherit</option>. Note that setting
-        this parameter might result in additional dependencies to be
-        added to the unit (see above).</para></listitem>
+        <para>This setting defaults to the value set with <option>DefaultStandardError=</option> in
+        <citerefentry><refentrytitle>systemd-system.conf</refentrytitle><manvolnum>5</manvolnum></citerefentry>, which
+        defaults to <option>inherit</option>. Note that setting this parameter might result in additional dependencies
+        to be added to the unit (see above).</para></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>StandardInputText=</varname></term>
+        <term><varname>StandardInputData=</varname></term>
+
+        <listitem><para>Configures arbitrary textual or binary data to pass via file descriptor 0 (STDIN) to the
+        executed processes. These settings have no effect unless <varname>StandardInput=</varname> is set to
+        <option>data</option>. Use this option to embedd process input data directly in the unit file.</para>
```

Comment:
Hmm, if StandardInput=null is specified and data is configured, it'll get upgraded automatically, no?

Reply:
yeah, but I didn't want to document this. I am not quite liking the concept of "null" meaning "usually null, but sometimes something else". I think we need to rework this later on, so that we have a special value for standard input which means "unset", and which is then either upgraded to "stdin" or to "null" (or whatever else) after loading. But i didn't want to add even more to this PR, hence I figured I'll do this in a later PR.

Hence, I'd consider it an undocumented trick that StandardInput=null + StandardInputText=foo actually effectively means StandardInput=data. I'd like to change this later on so that StandardInput=null actually means what it says...